packages/driver/src/cy/commands/actions/select.ts

```diff
@@ -279,6 +299,15 @@ export default (Commands, Cypress, cy) => {
               options.$el[0].selectedIndex = selectedIndex
             }
 
+            // https://github.com/cypress-io/cypress/issues/19494
```

Comment:
So I'm a bit confused here why we're using state() to store values. Would not a simpler implementation be to just store the actually selected DOM index beforehand, then compare to see if it's the same DOM element after selecting the new one? Eg:

```
const oldValue = options.$el[0].selectedIndex

options.$el.val(values)
if (notAllUniqueValues) { ... }

if (options.$el[0].selectedIndex === oldValue) {
  return;
}
```

It seems like this more cleanly encapsulates the desired behavior - "if the selected element hasn't changed, don't fire any events." Not sure why we need Cypress state here?

Reply:
As an example of a problem with the current approach, consider:
```
  <select class='ice-cream' name='ice-cream'>
    <option value=''>Select One …</option>
    <option value='chocolate' selected>Chocolate</option>
  </select>

        cy.visit('fixtures/select-event-counter.html')
        cy.get('.ice-cream').select('Chocolate')
        cy.get('#change-result').should('not.have.text', 'Number of times onChange event fired: 1')
        cy.get('#input-result').should('not.have.text', 'Number of times input event fired: 1')
```

This fails because the change event fires, which doesn't match browser behavior. No change should fire because chocolate had the "selected" attribute, so was selected when the fixture opened (and cy.select() doesn't change the selected element).

Reply:
We might also run into inconsistencies if those elements in state are detached between selections.

Reply:
Thank you. Your solution is much better. I changed the code in that way. I also added your test case.